weed/s3api/s3api_object_handlers.go

```diff
@@ -164,6 +177,12 @@ func (s3a *S3ApiServer) GetObjectHandler(w http.ResponseWriter, r *http.Request)
 	bucket, object := s3_constants.GetBucketAndObject(r)
 	glog.V(3).Infof("GetObjectHandler %s %s", bucket, object)
 
+	errCode := s3a.checkAccessForReadObject(r, bucket, object)
```

Comment:
It is reading the object metadata here, doubling the metadata read latency. not acceptable.

Reply:
Object access validation is now offloaded to the file layer; 
at `weed/server/filer_server_handlers_read.go:87`:
```go
func (fs *FilerServer) GetOrHeadHandler(w http.ResponseWriter, r *http.Request) {

	path := r.URL.Path
	isForDirectory := strings.HasSuffix(path, "/")
	if isForDirectory && len(path) > 1 {
		path = path[:len(path)-1]
	}

	entry, err := fs.filer.FindEntry(context.Background(), util.FullPath(path))
	if err != nil {
		if path == "/" {
			fs.listDirectoryHandler(w, r)
			return
		}
		if err == filer_pb.ErrNotFound {
			glog.V(2).Infof("Not found %s: %v", path, err)
			stats.FilerRequestCounter.WithLabelValues(stats.ErrorReadNotFound).Inc()
			w.WriteHeader(http.StatusNotFound)
		} else {
			glog.Errorf("Internal %s: %v", path, err)
			stats.FilerRequestCounter.WithLabelValues(stats.ErrorReadInternal).Inc()
			w.WriteHeader(http.StatusInternalServerError)
		}
		return
	}

	//s3 acl offload to filer
	offloadHeaderBucketOwner := r.Header.Get(s3_constants.X_SeaweedFS_Header_Amz_Bucket_Owner_Id)
	if len(offloadHeaderBucketOwner) > 0 {
		if statusCode, ok := s3acl.CheckObjectAccessForReadObject(r, w, entry, offloadHeaderBucketOwner); !ok {
			w.WriteHeader(statusCode)
			return
		}
	}
        ......
```
to resolve package cycle references, create separate packages for acl and accounts
<img width="231" alt="image" src="https://user-images.githubusercontent.com/48707349/194272695-bf9073a2-b116-4bde-9413-99e553cc56a1.png">